crates/ruff_linter/src/rules/pep8_naming/settings.rs

```diff
@@ -84,3 +64,127 @@ impl Error for SettingsError {
+
+        Ok(IgnoreNames::UserProvided(patterns))
+    }
+
+    /// Convert the [`IgnoreNames`] into a vector of [`IdentifierPattern`]s.
+    pub fn into_vec(self) -> Vec<IdentifierPattern> {
+        match self {
+            IgnoreNames::Default => vec![
+                IdentifierPattern::new("setUp").unwrap(),
+                IdentifierPattern::new("tearDown").unwrap(),
+                IdentifierPattern::new("setUpClass").unwrap(),
+                IdentifierPattern::new("tearDownClass").unwrap(),
+                IdentifierPattern::new("setUpModule").unwrap(),
+                IdentifierPattern::new("tearDownModule").unwrap(),
+                IdentifierPattern::new("asyncSetUp").unwrap(),
+                IdentifierPattern::new("asyncTearDown").unwrap(),
+                IdentifierPattern::new("setUpTestData").unwrap(),
+                IdentifierPattern::new("failureException").unwrap(),
+                IdentifierPattern::new("longMessage").unwrap(),
+                IdentifierPattern::new("maxDiff").unwrap(),
+            ],
+            IgnoreNames::UserProvided(patterns) => patterns,
+        }
+    }
+
+    /// Returns `true` if the given name matches any of the ignored patterns.
+    pub fn matches(&self, name: &str) -> bool {
+        match self {
+            IgnoreNames::Default => matches!(
+                name,
+                "setUp"
+                    | "tearDown"
+                    | "setUpClass"
+                    | "tearDownClass"
+                    | "setUpModule"
+                    | "tearDownModule"
+                    | "asyncSetUp"
+                    | "asyncTearDown"
+                    | "setUpTestData"
+                    | "failureException"
+                    | "longMessage"
+                    | "maxDiff"
+            ),
+            IgnoreNames::UserProvided(patterns) => {
+                patterns.iter().any(|pattern| pattern.matches(name))
+            }
+        }
+    }
+}
+
+impl fmt::Display for IgnoreNames {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            IgnoreNames::Default => {
+                writeln!(f, "[")?;
+                for elem in [
+                    "setUp",
+                    "tearDown",
+                    "setUpClass",
+                    "tearDownClass",
+                    "setUpModule",
+                    "tearDownModule",
+                    "asyncSetUp",
+                    "asyncTearDown",
+                    "setUpTestData",
+                    "failureException",
+                    "longMessage",
+                    "maxDiff",
+                ] {
+                    writeln!(f, "\t{elem},")?;
+                }
+                write!(f, "]")?;
+            }
+            IgnoreNames::UserProvided(patterns) => {
+                writeln!(f, "[")?;
+                for elem in patterns {
+                    writeln!(f, "\t{elem},")?;
+                }
+                write!(f, "]")?;
```

Comment:
\cc @snowsignal - in case you had ideas on this. I created a wrapper type with an enum variant that _can_ contain a vector, so had to (sorta) reimplement our settings-array formatting in the display.